storagemarket/impl/client.go

```diff
@@ -301,6 +299,34 @@ func (c *Client) SubscribeToEvents(subscriber storagemarket.ClientSubscriber) sh
 	return shared.Unsubscribe(c.pubSub.Subscribe(subscriber))
 }
 
+func (c *Client) restartDeals() error {
+	var deals []storagemarket.ClientDeal
+	err := c.statemachines.List(&deals)
+	if err != nil {
+		return err
+	}
+
+	for _, deal := range deals {
+		if c.statemachines.IsTerminated(deal) {
+			continue
+		}
+
+		s, err := c.net.NewDealStream(deal.Miner)
```

Comment:
So, both ClientDeal & MinerDeal have a value called ConnectionClosed. This is set to true when all the communication over libp2p that happens between provider/miner has happened (basically happens around the time of publishing). If ConnectionClosed is true, you should SKIP opening and adding the stream here.

Reply:
Oh, I didn't realize that's what `ConnectionClosed` was, will put in that check.